packages/synapse-react-client/src/components/EntityHeaderTable/EntityHeaderTable.tsx

```diff
@@ -0,0 +1,542 @@
+      return {
+        targetId: entityHeader.id,
+      }
+    })
+    updateRefsInState(newRowRefs)
+  }
+  const selectableTypes = [
+    EntityType.PROJECT,
+    EntityType.FOLDER,
+    EntityType.FILE,
+    EntityType.TABLE,
+    EntityType.LINK,
+    EntityType.ENTITY_VIEW,
+    EntityType.DOCKER_REPO,
+    EntityType.SUBMISSION_VIEW,
+    EntityType.DATASET,
+    EntityType.DATASET_COLLECTION,
+    EntityType.MATERIALIZED_VIEW,
+    EntityType.VIRTUAL_TABLE,
+  ]
+
+  const isSelection = selectionCount > 0
+  const totalRowCount = data.length
+  const filteredRowCount = table.getPrePaginationRowModel().rows.length
+  return (
+    <div>
+      <Box
+        sx={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          padding: '12px 10px 10px 5px',
+        }}
+      >
+        <div>
+          {totalRowCount > UNMANAGEABLE_SUBJECT_COUNT && (
+            <Typography variant="body1" sx={{ marginBottom: '10px' }}>
+              {totalRowCount} Entities{' '}
+              {filteredRowCount < totalRowCount
+                ? `(${filteredRowCount} visible)`
+                : ''}
+              {isSelection && <span>{` (${selectionCount} selected)`}</span>}
+            </Typography>
+          )}
+        </div>
+        <div>
+          {isEditable && (
+            <Button
+              variant="contained"
+              disabled={!isSelection}
+              onClick={onRemoveFromAR}
+            >
+              {removeSelectedRowsButtonText}
+            </Button>
+          )}
+        </div>
+      </Box>
+      {totalRowCount > 0 && (
+        <Box
+          sx={{
+            overflow: 'auto',
+            maxHeight: '250px',
+            paddingLeft: '2px',
+            th: {
+              backgroundColor: '#eee',
+              zIndex: 100,
+            },
+          }}
+        >
+          <table style={{ borderCollapse: 'collapse', width: '100%' }}>
+            <thead>
+              {table.getHeaderGroups().map(headerGroup => (
+                <tr key={headerGroup.id}>
+                  {headerGroup.headers.map(header => {
+                    let columnSize: string = '5%'
+                    switch (header.id) {
+                      case 'name':
+                        columnSize = '50%'
+                        break
+                      case 'id':
+                        columnSize = '22%'
+                        break
+                      case 'type':
+                        columnSize = '22%'
+                        break
+                      default:
+                        break
+                    }
+                    return (
+                      <th
+                        key={header.id}
+                        colSpan={header.colSpan}
+                        style={{
+                          width: columnSize,
+                          position: 'sticky',
+                          top: '0px',
+                          background: '#fff',
```

Comment:
Think this might work to prevent hard-coding (in my dreams we can one day justify a dark theme)
```suggestion
                          background: 'background.default',
```